website/project/model.py

```diff
@@ -1317,6 +1337,26 @@ def has_addon_on_children(self, addon):
 
         return False
 
+    def merge_public_files(self, node):
+
+        if not self.is_public_files_collection:
+            raise NodeStateError('must be Public Files collection to merge')
+
+        from website.files.models.osfstorage import OsfStorageFile
+
+        primaryUserFiles = OsfStorageFile.find(Q('node', 'eq', self) & Q('title', 'ne', 'Public Files'))
```

Comment:
Underscoring for database fields of the node object as in "self.is_public_files_collection" seems to be a convention as well though. Otherwise unsure how this doesn't fit.